### ISSUE TITLE
Consider default plugins for more parity with Node

### ISSUE THREAD

**Issue by sholladay** [rollup/rollup#1047]
Hello and thanks for the great work. `rollup` has treated me very well.

I would like to propose that the functionality gained via [`rollup-plugin-json`](https://github.com/rollup/rollup-plugin-json) and maybe [`rollup-plugin-node-resolve`](https://github.com/rollup/rollup-plugin-node-resolve) be included by default.

To be clear, since this _sounds_ like a slippery slope, what I'm really suggesting is that `rollup` have some parity with Node itself for compatibility with the wider ecosystem. By rallying around the same set of default file extensions (and Node really sets the bar here), interoperability becomes easier.

Part of what spurred this is [consistent linting](https://github.com/sindresorhus/xo/commit/ae7451cd047d6c18539bb004ec9c9ed1210b8659#commitcomment-19386652). Currently the JSON plugin requires using a `.json` extension, otherwise it can't find the file. This is unexpected and that brings up the point that _it should just work like Node_. I've been using the JSON plugin forever and some various hacks to get dependencies from `node_modules`. I only just today discovered the node-resolve plugin. So I think it's worth asking if the current design is worth the mental burden, even if there is a certain technical elegance to it.

I feel that I should be able to do this out of the box:

``` js
// Get package.json
import pkg from './package';
console.log(pkg.name, ':', pkg.version);
```

**Comment by jhabdas** [comment]
> To be clear, since this sounds like a slippery slope, what I'm really suggesting is that rollup have some parity with Node itself for compatibility with the wider ecosystem.

Agree. Slippery slope. ES Modules and Node have different goals. I'm :-1:.

**Comment by guybedford** [comment]
Closing as this has been inactive for a while and newer relevant issues can track these concerns. Happy to continue discussion as well here too.

**Comment by sholladay** [comment]
It's inactive because I never received an official response.from a maintainer.

Any thoughts on this, @guybedford? I still feel everything I stated here is just as relevant today. Rollup is used via Node, is linted by Node tooling, and the aspects where they diverge are just plain confusing and unintuitive.

If anything, I feel that this is more prudent now than back in 2016, because now we can confidently say that Node's resolve algorithm and JSON support still applies to Node's ESM environment.

**Comment by Rich-Harris** [comment]
See [here](https://github.com/rollup/rollup/issues/1555#issuecomment-322862209) for some thoughts on the subject

**Comment by lukastaegert** [comment]
Also, there is this suggestion that I think would also be a solution to your problem: https://github.com/rollup/rollup/issues/1895#issuecomment-359032598

Note that rollup is also used outside the node context, e.g. in the browser by our REPL.

**Comment by sholladay** [comment]
@Rich-Harris thanks for the link. The philosophy outlined in https://github.com/rollup/rollup/issues/1555#issuecomment-322862209 is exactly what I was referring to when I said:

> I think it's worth asking if the current design is worth the mental burden, even if there is a certain technical elegance to it.

In other words, I appreciate the technical arguments for having these plugins be entirely separate, especially from the maintainer's point of view. I just don't think it's good for users.

@lukastaegert a `--node` option would be a nice baby step. And maybe adding it will help make it clear that 95% of users will almost always use it. I predict that if you add an option like that, blog posts and tutorials everywhere will be updated to recommend it. I don't think they'll even bother to mention a use case without it. Your browser REPL is the edge case, not Node.

I ❤️ Rollup and just want its default experience to make people happy, so that more people will use it.

**Comment by Rich-Harris** [comment]
Thanks @sholladay, I definitely get where you're coming from, and I'll readily concede that my own perspective is somewhat warped by having worked on the codebase. The two sticking points for me are

1. the slippery slope argument is real — surely `--node` implies the `commonjs` plugin? And probably `node-builtins` etc. And before you know it we're shimming `process.nextTick`...
2. extensionless imports are just a really bad idea, and I get very grumpy about them!

In other words, I'm not strongly opposed to the idea, I just want to avoid inadvertently creating *more* complexity and confusion instead of less. For example we'd have to explain in the documentation why JSON is converted (even though browser native module loaders won't load it) and CommonJS isn't.
